guildwars2/account.py

```diff
@@ -347,24 +347,50 @@ def refined_armor_scan(a, b=ids_refined_envoy_armor):
             return await self.error_handler(ctx, e)
 
         def check(item):
-            return item is not None and item["id"] in choice["ids"]
+            if item is None:
+                return 0
+            if item["id"] in choice["ids"]:
+                if item.get("count") is not None:
+                    return item["count"]
+                return 1
+            if choice.get("isUpgrade") is None:
+                return 0
+
+            def countUpgrades(iteratable):
```

Comment:
snake_case for function names, typo in argument name, internal def's go at the top of the parent function usually. Function can be shortened as:
```py
return sum(1 for i in iterable if i in choice["ids"])
```